Core/Src/main.c

```diff
@@ -61,34 +72,44 @@ int main()
 
   RCC_Init();
   GPIO_Init();
-  UART1_Init();
   I2C1_Init();
+  UART1_Init(&uart1.uart);
```

Comment:
I would suggest to create a separate program module, that may be called `uart_commander` or something like that. Place these initializations of the ring buffer and uart to that new program module